File: tests/fixtures/sample_project_java/src/main/java/com/example/auth/JwtUtils.java
package com.example.auth;

/** JWT signing and signature verification helpers. */
public class JwtUtils {
    /** Produce a signed JWT string from a subject claim. */
    public static String encodeToken(String subject) {
        return sign("sub=" + subject);
    }

    /** Verify a JWT signature and return the payload if valid. */
    public static String decodeToken(String token) {
        if (token.startsWith("jwt.")) {
            return token.substring(4);
        }
        return null;
    }

    /** Sign a raw payload (internal helper). */
    private static String sign(String payload) {
        return "jwt." + payload;
    }
}

File: tests/fixtures/sample_project_csharp/src/Auth/JwtUtils.cs
namespace Acme.Auth;

/// <summary>JWT signing and signature verification helpers.</summary>
public class JwtUtils
{
    /// <summary>Produce a signed JWT string from a subject claim.</summary>
    public static string EncodeToken(string subject)
    {
        return Sign("sub=" + subject);
    }

    /// <summary>Verify a JWT signature and return the payload if valid.</summary>
    public static string DecodeToken(string token)
    {
        if (token.StartsWith("jwt."))
        {
            return token.Substring(4);
        }
        return null;
    }

    /// <summary>Sign a raw payload (internal helper).</summary>
    private static string Sign(string payload)
    {
        return "jwt." + payload;
    }
}
